pkg/ccl/backupccl/restore_job.go

```diff
@@ -317,7 +317,8 @@ func restore(
 		return emptyRowCount, err
 	}
 
-	on231 := clusterversion.TODO_Delete_V23_1.Version().LessEq(job.Payload().CreationClusterVersion)
+	ver := job.Payload().CreationClusterVersion
+	on231 := ver.Major > 23 || (ver.Major == 23 && ver.Minor >= 1)
```

Comment:
I _think_ that with MinVersion set to 24_1 the only way we were get here is if the user has started their restore with `unsafe_restore_incompatible_version`.   Doesn't need to happen in this PR, but seems like we might be able to just remove this now. cc @msbutler

Reply:
Adding just a TODO in this PR.